app/IOSGateway.m

```diff
@@ -0,0 +1,121 @@
+//
+//  IOSGateway.m
+//  iSH
+//
+//  Created by Miguel Vanhove on 15/02/2019.
+//
+
+#import "IOSGateway.h"
+#import "IACManager.h"
+#import "UIApplication+OpenURL.h"
+
+static IOSGateway *iOSGateway = nil;
+
+@interface IOSGateway ()
+{
+}
+
+@property (copy, nonatomic) NSData *iacResult;
+
+@end
+
+@implementation IOSGateway
+{
+}
+
++ (IOSGateway *_Nullable)sharedSession
+{
+    if (iOSGateway == nil) {
+        iOSGateway = [[IOSGateway alloc] init];
+    }
+
+    return iOSGateway;
+}
+
+- (id)init
+{
+    self = [super init];
+    if (self != nil) {
```

Comment:
I like to be cute and stick the setting of `self` and check for `nil` in this condition, but again, this is a stylistic choice.